File: 017/cpp/src/main.cpp
//
// main.cpp
// Project Euler - Problem 17
//
// Jordan Sola 2020 - MIT License

#include <iostream>
#include <string>

using std::cin;
using std::cout;
using std::endl;
using std::string;

const string empty = "";

const string ones[] = {empty, "One", "Two", "Three", "Four", "Five",
                    "Six", "Seven", "Eight", "Nine", "Ten",
                    "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen",
                    "Sixteen", "Seventeen", "Eighteen", "Nineteen"};

const string tens[] = {empty, empty, "Twenty", "Thirty", "Forty", "Fifty",
                    "Sixty", "Seventy", "Eighty", "Ninety"};

string hundred = "Hundred";
string thousand = "Thousand";
string million = "Million";
string billion = "Billion";

string ConvertTwoDigit(int num)
{
    if (num == 0)
    {
        return empty;
    }
    else if (num > 19)
    {
        return tens[num / 10] + ones[num % 10];
    }
    else
    {
        return ones[num];
    }
}

string Num2Word(unsigned long long int num, string suffix)
{

}

int main()
{
    unsigned long long int max = 1000;

    for (int i = 1; i < max + 1; i++)
    {
        cout << ConvertTwoDigit(i) << endl;
    }

    return 0;
}

File: 017/python/main.py
#
# main.py
# Project Euler - Problem 17
#
# Jordan Sola 2020 - MIT License

from num2words import num2words

length = 1000
words = []

for i in range(length):
    # we're not counting zero so we need to start our words at one
    words.append(num2words(i + 1))

# Remove spaces and dashes, they cannot be counted.
for i in range(length):
    words[i] = words[i].replace(' ', '')
    words[i] = words[i].replace('-', '')

total_letters = 0

for i in words:
    total_letters += len(i)

print(total_letters)